python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/client.py

```diff
@@ -355,18 +364,46 @@ def list_run_artifacts(
             )["data"],
         )
 
-    def get_run_artifact(self, run_id: int, path: str) -> Mapping[str, Any]:
+    def get_run_artifact(
+        self, run_id: int, path: str, return_text: bool = False
+    ) -> Mapping[str, Any]:
         """Retrieves an artifact at the given path for a given dbt Cloud Run.
 
+        Args:
+            run_id (int): The dbt Cloud Run ID.
+            path (str): The path to the artifact (e.g., "logs/dbt.log", "run_results.json").
+            return_text (bool): If True, returns the raw text content instead of parsing as JSON.
+                Useful for log files.
+
         Returns:
-            Dict[str, Any]: Parsed json data representing the API response.
+            Dict[str, Any]: Parsed json data representing the API response, or the raw text if
+                return_text is True.
         """
-        return self._make_request(
-            method="get",
-            endpoint=f"runs/{run_id}/artifacts/{path}",
-            base_url=self.api_v2_url,
-            session_attr="_get_artifact_session",
-        )
+        url = f"{self.api_v2_url}/runs/{run_id}/artifacts/{path}"
+
+        num_retries = 0
+        while True:
```

Comment:
why is this change necessary? seems like it's basically the same logic as the existing _make_request method already has, so if there's something lacking about that method i'd prefer if we just updated that instead of replicating it here

Reply:
not necessary -- wasn't sure the best way to handle that `_make_requests()` now might return text or JSON while keeping the return type as a dictionary. Not 100% happy with the current solution I just put in (trying to parse the JSON and then adding "text"), but it looks like the http response header from dbt cloud doesn't always include an "application/json" for json so I'm not sure if there's a better way